### PR TITLE
Add dialog for adding draft posts to sequences

### PR DESCRIPTION
**Pull request by michaelkeenan**
This PR enables people to add *draft* posts to sequences. Previously, people could only add published posts to sequences, because the UI used the Posts index to find posts, and the Posts index doesn't contain draft posts.

I've added an Add Draft Post button to the section where you add/remove posts to the sequence – see it in the bottom right in this image:

Screenshot 2023-02-15 at 12 45 35 PM

Clicking the button opens a dialog box that lists your drafts (and any drafts that are shared with you, if you toggle that option). It has the same filtering and sorting options as on the My Drafts page. When you click a draft post, it becomes gray-disabled, and is added to the sequence.

Screenshot 2023-02-15 at 12 45 43 PM

One place of possible concern is the extra parameters I've added to PostsItemWrapper. It's a component to display line items describing posts, like PostsItem2 but much simpler. It used to be quite simple:

`const PostsItemWrapper = ({documentId, classes, removeItem}`

Now it's less simple:

`const PostsItemWrapper = ({documentId, classes, addItem, removeItem, disabled = false, simpleAuthor = false, draggable = true}`

I think all these options make sense, but the road to huge unwieldy objects is paved with tiny changes, each individually justifiable.

(Why use a dialog? The following is a discussion of that UI choice, which is only important if you care about that choice, so I suggest skipping it if that's not important to you, as it might be hard to follow.

Rather than opening a dialog to select draft posts, we'd previously talked about adding a list of draft posts under the current Add/Remove posts form. It'd be kind of a messy look, with a list of sequence posts, then Submit/Cancel buttons (because they're part of the Vulcan form so they're not movable), then draft posts listed below. The draft post list below would be irrelevant clutter for most people. Using a dialog box hides that away cleanly.

But the real reason I used a dialog is that the current Add/Remove posts section is a Vulcan form with state that's hard to keep consistent with the draft posts list. The new section that adds draft posts needs to keep track of the sequence's posts, so that it can append the new draft post's postId to that list and replace it. Upon making this change, the Vulcan form notices the changed post list and updates, replacing its list of posts with the new list that includes the new draft post. That's good, unless you've already made changes in the Vulcan form (like adding or removing posts, or changing their order), in which case those changes are overwritten, because the new UI section only knows the postIds list as it was when the page was loaded. I eventually discovered that it's possible to keep track of the Vulcan form's changes to the postIds via its changeCallback, so it's not impossible to keep the state consistent that way, but that approach feels hacky and doesn't exist elsewhere in the codebase. I like the dialog box UI better anyway—it's less clutter.)

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1203981454226119) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by Raemon** [comment]
I don't have objections to the dialog. I definitely agree that having the draft-posts UI only appear after clicking a button is better than appearing by default. 

Looking at PostsItemWrapper.... actually I think it's fine for it to receive the changes you're proposing, and mostly... I think PostsItemWrapper is a silly misleading name for the component. (But, that doesn't seem like your job to fix right now)

I am maybe concretely confused about why you need to pass "simple" into the PostsUserAndCoauthors though.

**Comment by michaelkeenan** [comment]
"simple" is a weird name for that parameter. It makes the post author display "simply", i.e. not as a link. I needed that because I wanted the whole post row to be the hitbox for adding it to the sequence posts, but if you happened to click on the author name when it wasn't "simple", you'd navigate to the author's user page.

File: packages/lesswrong/components/sequences/SequenceDraftsList.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    terms,
+    collectionName: "Posts",
+    fragmentName: 'PostsList',
+    fetchPolicy: 'cache-and-network',
+    nextFetchPolicy: "cache-first",
+  });
+
+  if (!currentUser) return null
+
+  const currentSorting = terms.sortDraftsBy || "lastModified"
```
Comment (b0b3rt) [inline packages/lesswrong/components/sequences/SequenceDraftsList.tsx:56]: I don't think the `|| "lastModified"` is necessary, since that's already happening in the declaration of `terms`

Comment (b0b3rt) [reply packages/lesswrong/components/sequences/SequenceDraftsList.tsx:56]: Maybe this can just be declared (above `terms`) as `const currentSorting = query.sortDraftsBy ?? query.view ?? currentUser?.draftsListSorting ?? "lastModified";`, and then `terms` just has `sortDraftsBy: currentSorting`?

Comment (michaelkeenan) [reply packages/lesswrong/components/sequences/SequenceDraftsList.tsx]: I've made this change in SequenceDraftsList.tsx, and also in DraftsList.tsx (the "My Drafts" component, upon which SequenceDraftsList is based).

**Review by b0b3rt — APPROVED** [review]
